dev-docs/source/MVPTutorial/Matplotlib.rst

```diff
@@ -39,36 +42,33 @@ the plot and creating an empty plot (no data).
 
 .. code-block:: python
 
-    class PlotView(QtWidgets.QWidget):
-        def __init__(self, parent=None):
+    class PlotView(QWidget):
+        def __init__(self, parent: Union[QWidget, None]=None):
             super().__init__(parent)
 
-            self.figure = plt.figure()
-            grid = QtWidgets.QVBoxLayout(self)
+            self._figure = plt.figure()
+            grid = QVBoxLayout(self)
             self.draw()
-            self.canvas = self.getWidget()
-            grid.addWidget(self.canvas)
+            self._canvas = FigureCanvas(self._figure)
+            grid.addWidget(self._canvas)
             self.setLayout(grid)
 
-        def draw(self):
-            ax = self.figure.add_subplot(111)
+        def draw(self) -> Axes:
+            ax = self._figure.add_subplot(111)
             ax.clear()
             ax.set_xlim([0.0, 10.5])
             ax.set_ylim([-1.05, 1.05])
             ax.set_xlabel("time ($s$)")
             ax.set_ylabel("$f(t)$")
             return ax
 
-        def getWidget(self):
-            return FigureCanvas(self.figure)
-
-        def addData(self, xvalues, yvalues, grid_lines, colour, marker):
+        def add_data(self, xvalues: ndarray, yvalues: ndarray, visible: bool, colour: str, marker: str) -> None:
```

Comment:
This function is probably better renamed as plot_data? I say because it doesn't add data, just clears the axes and plots new one.